### PR TITLE
Fix linker error in Android builds on Linux 

### PR DESCRIPTION
**Pull request by hjanetzek**
## What does this PR do?

Fixing linker error  by not linking libraries from ${LY_NDK_ABI_ROOT}/usr/lib since LY_NDK_ABI_ROOT is never set.
`'ld: error: --fix-cortex-a53-843419 is only supported on AArch64 targets' ` 

## How was this PR tested?

Android build on linux ran succesfully.

### CODE REVIEW

File: cmake/Platform/Android/Configurations_android.cmake [hunk 1]
```diff
@@ -56,7 +56,6 @@ if(CMAKE_CXX_COMPILER_ID STREQUAL "Clang")
 
             -u ANativeActivity_onCreate
 
-            -L${LY_NDK_ABI_ROOT}/usr/lib
```
Comment (lumberyard-employee-dm) [inline cmake/Platform/Android/Configurations_android.cmake:59]: So the NDK on Windows has this set, but not NDK on Linux?

Comment (hjanetzek) [reply cmake/Platform/Android/Configurations_android.cmake:59]: I guess a regular Windows system does not have /usr/lib containing conflicting files

Comment (hjanetzek) [reply cmake/Platform/Android/Configurations_android.cmake:59]: On Windows this line should have no effect at all as it likely sets a non-existent library path, but it breaks the build on linux and probably macos as well

Comment (hjanetzek) [reply cmake/Platform/Android/Configurations_android.cmake:59]: https://github.com/o3de/o3de/search?q=LY_NDK_ABI_ROOT this is the single occurrence of LY_NDK_ABI_ROOT so the removed line should expand to `-L/usr/lib` on Windows as well and not to a path within NDK

Comment (lumberyard-employee-dm) [reply cmake/Platform/Android/Configurations_android.cmake:59]: It also appears that `LY_NDK_SRC_ROOT` is also never set

**Review by lumberyard-employee-dm — APPROVED** [review]

File: cmake/Platform/Android/Configurations_android.cmake (hunk 1 above)
Comment (hjanetzek) [reply cmake/Platform/Android/Configurations_android.cmake:59]: sure!

**Review by gadams3 — APPROVED** [review]

**Review by burelc-amzn — APPROVED** [review]
These 2 variables used to be set by the `cmake/Platform/Android/Toolchain_android.cmake` file, but were removed quite some time ago.

**Review by cgalvan — APPROVED** [review]

**Review by spham-amzn — APPROVED** [review]

**Pull request merged**